### ISSUE TITLE
`ember/no-runloop` brings false positive error with function name `hasOwnProperty`

### ISSUE THREAD

**Issue by mkszepp** [ember-cli/eslint-plugin-ember#2150]
After cleanup of all  `ember/runloops`, i have removed the rule `ember/no-runloop`.
Now i'm getting a false positive error.

Example:

```js
import Route from '@ember/routing/route';
import { service } from '@ember/service';

export default class MyRoute extends Route {
  @service router;

  setupController(controller) {
    super.setupController(controller);

    hasOwnProperty();
  }
}

function hasOwnProperty() {
  return 'hello';
}
``` 

![grafik](https://github.com/ember-cli/eslint-plugin-ember/assets/19845290/0d7c43bd-8832-448d-abbb-9b26e2ce1ec3)

While the lint rules brings wrong errors, there is necessary to disable the rule, because there is not allowed to set `hasOwnproperty` in `allowedList`.

```js
// not allowed - error "Value "hasOwnProperty" should be equal to one of the allowed values."
 'ember/no-runloop': [
      'error',
      {
        allowList: ['hasOwnProperty'],
      },
    ],
```

Version:
- eslint-plugin-ember: 12.1.1 (also tested with 12.0.2)
- Ember: 5.8
- Ember cli: 5.8.1

**Comment by basz** [comment]
i'm seeying a false positive too at line 34 using 12.1.1

```
const svg = await toString(data, options);
```

```
import { assert } from '@ember/debug';
import Service from '@ember/service';

import { toDataURL, toString } from 'qrcode';

import type { QRCodeToDataURLOptionsJpegWebp, QRCodeToDataURLOptionsOther, QRCodeToStringOptions } from 'qrcode';

export default class RendererService extends Service {
  async renderDataUrl(data: string, options: QRCodeToStringOptions | QRCodeToDataURLOptionsJpegWebp | QRCodeToDataURLOptionsOther = {}): Promise<string> {
    if (!data.length) {
      return '';
    }

    if (!options?.type) {
      options.type = 'image/png';
    }

    if (!options?.errorCorrectionLevel) {
      options.errorCorrectionLevel = 'L';
    }

    if (!options?.margin) {
      options.margin = 0;
    }

    assert(`Attribute 'type' must be one of 'image/png', 'svg', 'image/jpeg'`, ['image/png', 'svg', 'image/jpeg'].includes(options.type));
    assert(`Attribute 'data' must be string...`, 'string' === typeof data);
    assert(`Attribute 'errorCorrectionLevel' must be string`, 'string' === typeof options.errorCorrectionLevel);
    assert(`Attribute 'errorCorrectionLevel' must be one of 'L', 'M', 'Q' or 'H'`, ['L', 'M', 'Q', 'H'].includes(options.errorCorrectionLevel));
    //    assert(`Attribute 'quality' must be a number`, options?.rendererOpts?.quality !== undefined ? typeof options?.rendererOpts?.quality === 'number' : true);

    switch (options.type) {
      case 'svg': {
        const svg = await toString(data, options);
        const blob = new Blob([svg], { type: 'image/svg+xml' });

        return URL.createObjectURL(blob);
      }
      case 'image/png':
      case 'image/jpeg':
        return await toDataURL(data, options);
      default:
        throw new Error('Unsupported QR Type');
    }
  }
}

declare module '@ember/service' {
  interface Registry {
    'qr-renderer': RendererService;
  }
}
```

**Comment by mkszepp** [comment]
Looks like all Object.prototype properties ( https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object_prototypes) are causing false positive error...

```js 
hasOwnProperty()
isPrototypeOf()
propertyIsEnumerable()
toLocaleString()
toString()
valueOf()
constructor()
```

**Comment by basz** [comment]
@mkszepp did you find a work around?
